### PR TITLE
Feature/net6 upgrade

### PR DESCRIPTION
**Pull request by TRoZZNL**
Contains the following major changes:

- Upgraded solution to .NET 6.0
- Included Preview handler in SendgridMailer
- Async everything, since Mediakiwi supports Async.
- Upgraded some nuget Packages in light of .NET 6.0

### CODE REVIEW

File: src/Sushi.MailTemplate.SendGrid/QueuePersister.cs [hunk 1]
```diff
@@ -16,53 +11,10 @@ public QueuePersister(string connectionString)
...
-            // Retrieve a reference to a queue
-            CloudQueue queue = queueClient.GetQueueReference(name);
-
-            //did we already call createIfNotExists for this container?
-            string cacheKey = QueueCachePrefix + storageAccount.Credentials.AccountName + name;
-
-            var cache = System.Runtime.Caching.MemoryCache.Default;
-            if (!cache.Contains(cacheKey))
-            {
-                queue.CreateIfNotExists();
```
Comment (RobinWitteman) [inline src/Sushi.MailTemplate.SendGrid/QueuePersister.cs:36]: CreateIfNotExists is removed, is this correct?

**Review by RobinWitteman — CHANGES_REQUESTED** [review]

File: src/Sushi.MailTemplate.MediaKiwi.Portal/Mailer.cs [hunk 2]
```diff
@@ -31,15 +29,15 @@ public Mailer(string emailStorageAccount, string emailBlobContainer, string emai
         /// <param name="mail"></param>
         /// <param name="emailTo"></param>
         /// <returns></returns>
-        public bool SendMailToQueue(Sushi.MailTemplate.Data.MailTemplate mail, string emailTo, Guid? customerGuid)
+        public async Task<bool> SendMailToQueue(Data.MailTemplate mail, string emailTo, Guid? customerGuid)
```
[the remark this answers is not in the document]
Comment (TRoZZNL) [reply src/Sushi.MailTemplate.MediaKiwi.Portal/Mailer.cs]: Agreed, is adjusted

File: src/Sushi.MailTemplate.SendGrid/BlobPersister.cs [hunk 3]
```diff
@@ -17,117 +13,12 @@ public BlobPersister(string connectionString)
...
-            var result = blobClient.GetContainerReference(name);
-
-            //did we already call createIfNotExists for this container?
-            string cacheKey = ContainerCachePrefix + StorageAccount.Credentials.AccountName + name;
-
-            var cache = System.Runtime.Caching.MemoryCache.Default;
-            if (!cache.Contains(cacheKey))
-            {
-                //create if not exists
-                if (result.CreateIfNotExists())
```
[the remark this answers is not in the document]
Comment (TRoZZNL) [reply src/Sushi.MailTemplate.SendGrid/BlobPersister.cs:45]: re-added this code, was not supposed to be deleted

File: src/Sushi.MailTemplate.SendGrid/QueuePersister.cs (hunk 1 above)
Comment (TRoZZNL) [reply src/Sushi.MailTemplate.SendGrid/QueuePersister.cs:36]: re-added this code, was not supposed to be deleted

File: src/Sushi.MailTemplate.SendGrid/Mailer.cs [hunk 4]
```diff
@@ -212,20 +227,17 @@ public async Task<bool> SendMailFromBlobAsync(string id)
             }
 
             // get storage account
-            var storageAccount = Microsoft.WindowsAzure.Storage.CloudStorageAccount.Parse(EmailStorageAccount);
+            var storageAccount = new BlobPersister(EmailStorageAccount);
```
Comment (RobinWitteman) [inline src/Sushi.MailTemplate.SendGrid/Mailer.cs:230]:
```suggestion
            // get blob persister
            var blobPersister = new BlobPersister(EmailStorageAccount);
```

**Review by RobinWitteman — CHANGES_REQUESTED** [review]

**Review by RobinWitteman — APPROVED** [review]

**Pull request merged**